ui/pages/settings/networks-tab/networks-form/networks-form.js

```diff
@@ -284,6 +299,41 @@ const NetworksForm = ({
     isEditing,
   ]);
 
+  const newOrderNetworks = () => {
```

Comment:
whats the reason for sorting the networks here?

since they're only used to look up a network by chain id, maybe this won't be needed once we fully migrate to 1 network per chain?

Reply:
i think this line sorts the `nonTestNetworks` array. The sort function compares two elements (a and b) by creating keys keyA and keyB using their `chainId` and `rpcUrl`. It then uses the orderedIndexMap to determine their order based on the indices from `orderedNetworksList`.
which means for same network with same chainID and differents rpc, we will sort them.

Reply:
however this is not needed for the new network add/edit flow since we will group them , so i'm going to filter this by feature flag